server/src/main/java/org/elasticsearch/index/analysis/Analysis.java

```diff
@@ -351,7 +351,7 @@ public static Reader getReaderFromFile(Environment env, String filePath, String
 
     public static Reader getReaderFromIndex(String synonymsSet, SynonymsManagementAPIService synonymsManagementAPIService) {
         final PlainActionFuture<PagedResult<SynonymRule>> synonymsLoadingFuture = new PlainActionFuture<>();
-        synonymsManagementAPIService.getSynonymSetRules(synonymsSet, 0, 10_000, synonymsLoadingFuture);
+        synonymsManagementAPIService.getSynonymSetRules(synonymsSet, synonymsLoadingFuture);
```

Comment:
Retrieve all synonyms sets. It does not specify the maximum to allow the service to set that up depending on the index setting for bwc reasons.